app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -73,31 +76,34 @@ class FxaRepo(
         return accountManager.beginAuthenticationAsync()
     }
 
-    private inner class FirefoxAccountObserver : AccountObserver {
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    inner class FirefoxAccountObserver : AccountObserver {
+        /**
+         * The account profile is fetched asynchronously.
+         * There is no way to transition from [NOT_AUTHENTICATED] directly to [AUTHENTICATED_WITH_PROFILE];
+         * [AUTHENTICATED_NO_PROFILE] is always a state in between.
+         */
         override fun onAuthenticated(account: OAuthAccount) {
             logger.debug("onAuthenticated")
-            // todo: is this correct?
-            val nextState = if (accountManager.accountProfile() != null) {
-                AUTHENTICATED
-            } else {
-                AUTHENTICATED_NO_PROFILE
-            }
-            accountStateDontUseMeYet.onNext(nextState)
+            _accountState.onNext(AUTHENTICATED_NO_PROFILE)
         }
 
         override fun onAuthenticationProblems() {
             logger.debug("onAuthenticationProblems")
-            accountStateDontUseMeYet.onNext(NEEDS_REAUTHENTICATION)
+            _accountState.onNext(NEEDS_REAUTHENTICATION)
         }
 
         override fun onLoggedOut() {
             logger.debug("onLoggedOut")
-            accountStateDontUseMeYet.onNext(NOT_AUTHENTICATED)
+            _accountState.onNext(NOT_AUTHENTICATED)
         }
 
+        /**
+         * This is called when the profile is first fetched after sign-in.
+         */
         override fun onProfileUpdated(profile: Profile) {
```

Comment:
I don't think it matters in the current implementation but have we taken into account that this may be called multiple times? e.g. if the user changes their avatar, I'd expect this to be called.

Reply:
From looking at the FxA code, this only gets called once.

Reply:
It only gets called in one location but it's not guaranteed to only be called once. For example, the public method `FxaAccountManager.updateProfileAsync` queues a `FetchProfile` event, which will fetch the profile and queue a `FetchedProfile` event which will call `onProfileUpdated`.

@dnarcese Is the current code resilient to this being called multiple times?